lib/generator.js

```diff
@@ -642,9 +642,36 @@ class Generator {
     } catch (e) {
       this.templateConfig = {};
     }
+    await this.loadDefaultValues();
     await this.validateTemplateConfig();
   }
 
+  /**
+   * Loads default values of parameters from template config. If value was already set as parameter it will not be
+   * overrided.
+   * @private
+   */
+  async loadDefaultValues() {
+    const parameters = this.templateConfig.parameters;
+    const defaultValues = [];
+    Object.keys(parameters || {}).forEach(key => {
```

Comment:
We are already iterating over the parameters keys here: https://github.com/asyncapi/generator/blob/master/lib/generator.js#L664.

Can we reuse this loop instead of iterating again?

Reply:
It's in `validateTemplateConfig` method, different purposes. Btw i do not want to complicate the `validateTemplateConfig` method even more.

Reply:
code updated

Reply:
Awesome! Thanks for taking care!